api/src/wfl/module/covid.clj

```diff
@@ -152,6 +152,22 @@
                              {:workload workload})))
     (if-not (or stopped finished) (stop! workload (utc-now)) workload)))
 
+(defn ^:private retry-covid-workload
+  "Retry/resubmit the `workflows` managed by the `workload` and return the
+   workload that manages the new workflows."
+  [{:keys [finished id executor] :as workload} workflows]
+  (when-not finished
+    (throw (UserException. "Cannot retry workload before it's finished."
+                           {:workload workload})))
+  ;; TODO: validate executor and sink.
+  ;; Existing stage/validate-or-throw dispatches on :name.
+  ;; The workload's executor and sink have :type at this stage, not :name.
+  (executor/retry-executor! executor workflows)
+  (jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
+    (when-not (stage/done? executor)
+      (patch-workload tx workload {:finished nil :updated (utc-now)}))
```

Comment:
I got burned by this!

Initially, I revived the workload *before* the retry.  Then I noticed a race condition where the update loop might mark the workload as finished before the retry workflow records could be written.